providers/apis/binance/api_handler.go

```diff
@@ -106,12 +93,12 @@ func (h *APIHandler) ParseResponse(
 
 	for _, data := range result {
 		// Filter out the responses that are not expected.
-		ticker, ok := h.market.OffChainMap[data.Symbol]
+		ticker, ok := h.cache.FromOffChainTicker(data.Symbol)
```

Comment:
We're introducing a cache consistency problem here, no? If the cached tickers vary from the given tickers we can have an inconsistency. If we're caching all ticker state to avoid O(n) lookups, I'd either update the cache in `ParseResponse` or pass a set, instead of an array of `ProviderTickers` to `ParseResponse`

Reply:
CreateURL will always be called before ParseResponse but im fine moving the caching from CreateURL to here. seems like the safest option.